src/tpm.rs

```diff
@@ -572,40 +584,56 @@ pub fn run<'a>(
 
                 thread::sleep(RETRY_SLEEP);
             }
-            _ => break,
+
+            _ => break 'exec,
         }
     }
 
-    let return_output = output.stdout;
-    let return_code = output.status.code();
+    // preprocess execution result
+    let return_output = String::from_utf8(output.stdout).map_err(|e| {
```

Comment:
I don't see `return_output` being used by anything apart from incorrect return code error handling, so I don't see a reason to return it when nothing goes wrong. (Unless I'm missing something obvious.)

Reply:
`return_output` contains information that the caller function needs instead of the error output. That is because after the TPM command executed, the information goes to both the standard output and the file that was given depending on which command you were using.